yt_dlp/extractor/googledrive.py

```diff
@@ -289,55 +295,93 @@ def add_source_format(urlh):
+                end_pattern=_META_END_RE, contains_pattern=_ARRAY_RE, **kwargs)
+        elif hashval:
+            if not name:
+                name = f'webpage JSON metadata hash:{hashval}'
+            return self._search_json(
+                rf'''hash\s*?:\s*?(['"]){hashval}\1,[^\[]*?data:''', webpage, name, video_id,
+                end_pattern=_META_END_RE, contains_pattern=_ARRAY_RE, **kwargs)
+        return None
 
---{_BOUNDARY}
-'''
-
-    def _call_api(self, folder_id, key, data, **kwargs):
-        response = self._download_webpage(
-            'https://clients6.google.com/batch/drive/v2beta',
-            folder_id, data=data.encode(),
-            headers={
-                'Content-Type': 'text/plain;charset=UTF-8;',
-                'Origin': 'https://drive.google.com',
-            }, query={
-                '$ct': f'multipart/mixed; boundary="{self._BOUNDARY}"',
-                'key': key,
-            }, **kwargs)
-        return self._search_json('', response, 'api response', folder_id, **kwargs) or {}
+    def _real_extract(self, url):
+        def item_url_getter(item, video_id):
+            if not isinstance(item, list):
+                return None
+            available_IEs = [GoogleDriveFolderIE, GoogleDriveIE]  # subfolder or item
+            if 'application/vnd.google-apps.shortcut' in item:  # extract real link
+                entry_url = traverse_obj(
+                    item,
+                    (..., ..., lambda _, v: any(ie.suitable(v) for ie in available_IEs), any))
+            else:
+                entry_url = traverse_obj(
+                    item,
+                    (lambda _, v: any(ie.suitable(v) for ie in available_IEs), any))
+            if not entry_url:
+                return None
+            return self.url_result(entry_url, video_id=video_id, video_title=item[2])
 
-    def _get_folder_items(self, folder_id, key):
-        page_token = ''
-        while page_token is not None:
-            request = self._REQUEST.format(folder_id=folder_id, page_token=page_token, key=key)
-            page = self._call_api(folder_id, key, self._DATA % request)
-            yield from page['items']
-            page_token = page.get('nextPageToken')
+        folder_id = self._match_id(url) or 'my-drive'
+        headers = self.geo_verification_headers()
 
-    def _real_extract(self, url):
-        folder_id = self._match_id(url)
+        try:
+            webpage, urlh = self._download_webpage_handle(url, folder_id, headers=headers)
+        except ExtractorError as e:
+            if isinstance(e.cause, HTTPError):
+                if e.cause.status == 404:
+                    self.raise_no_formats(e.cause.msg)
+                elif e.cause.status == 403:
+                    self.raise_login_required('Access Denied!')
+            raise
+        if urllib.parse.urlparse(urlh.url).netloc == 'accounts.google.com':
```

Comment:
needs to handle redirection before this. if the passed domain name is `docs.google.com`, there will be a redirection to `drive.google.com` before redirecting to `accounts.google.com` if the folder is private.